### PR TITLE
[ADD] Encryption of sensitive data

### LINKED ISSUE #156: Secure secrets
- [x] Secret information such as access data should only be stored in the database in encrypted form.
- [x] Secrets should not be displayed in the frontend.

### PR DESCRIPTION
**Pull request by Plaenkler**
Closes #156

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -115,6 +122,10 @@ func loadFromEnv() error {
 	if resolver != "" {
 		config.Resolver = resolver
 	}
+	Cryptor, ok := os.LookupEnv("DDNS_Cryptor")
+	if ok && Cryptor != "" {
+		config.Cryptor = Cryptor
+	}
```
[the remark this answers is not in the document]
Comment (Plaenkler) [reply pkg/config/config.go:128]: Does not have to be user provided

File: pkg/config/config.go [hunk 2]
```diff
@@ -115,6 +122,10 @@ func loadFromEnv() error {
 	if resolver != "" {
 		config.Resolver = resolver
 	}
+	Cryptor, ok := os.LookupEnv("DDNS_Cryptor")
```
[the remark this answers is not in the document]
Comment (Plaenkler) [reply pkg/config/config.go]: Fixed in 493b404

File: README.md [hunk 3]
```diff
@@ -141,4 +148,5 @@ DDNS_INTERVAL=600
 DDNS_TOTP=false
 DDNS_PORT=80
 DDNS_RESOLVER=ipv4.example.com
+DDNS_Cryptor=29atdqljyqUVXNAuXltyng==
```
[the remark this answers is not in the document]
Comment (Plaenkler) [reply README.md]: Fixed in https://github.com/Plaenkler/DDNS-Updater/commit/493b404f29280b98743e689c358ec47c3b9f3623

No just an example

File: pkg/server/routes/web/index.go [hunk 4]
```diff
@@ -87,21 +91,45 @@ func ProvideIndex(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func formatParams(jobs []model.SyncJob) error {
+func sanatizeParams(jobs []model.SyncJob) error {
```
[the remark this answers is not in the document]
Comment (Plaenkler) [reply pkg/server/routes/web/index.go]: Fixed in 90a33b6f8056885825efda7b9fcef90015ec62a3

File: pkg/cipher/cipher.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	}
+	nonce := make([]byte, gcm.NonceSize())
+	_, err = io.ReadFull(rand.Reader, nonce)
+	if err != nil {
+		return "", fmt.Errorf("[cipher-Encrypt-3] encryption failed: %s", err)
+	}
+	return base64.StdEncoding.EncodeToString(gcm.Seal(nonce, nonce, []byte(plaintext), nil)), nil
+}
+
+func Decrypt(key string, ciphertext string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Plaenkler) [reply pkg/cipher/cipher.go]: Fixed in 61b1af2bf70a3d3c593f881c514d28398bcf10fa

**Review by Valentin-Kaiser — APPROVED** [review]

**Pull request merged**